src/dbManager.js

```diff
@@ -0,0 +1,161 @@
+const BN = require('bn.js')
+const level = require('level-mem')
+const rlp = require('rlp')
+const Block = require('ethereumjs-block')
+const Cache = require('./cache')
+const {
+  headsKey,
+  headHeaderKey,
+  headBlockKey,
+  hashToNumberKey,
+  numberToHashKey,
+  tdKey,
+  bodyKey,
+  headerKey
+} = require('./util')
+
+module.exports = class DBManager {
+  constructor (db, common) {
+    this._db = db
+    this._common = common
+    this._opts = {
+      keyEncoding: 'binary',
+      valueEncoding: 'binary'
+    }
+    this._cache = {
+      td: new Cache({ max: 1024 }),
+      header: new Cache({ max: 512 }),
+      body: new Cache({ max: 256 }),
+      numberToHash: new Cache({ max: 2048 }),
+      hashToNumber: new Cache({ max: 2048 })
+    }
+  }
+
+  getHeads () {
+    return this._db.get(headsKey, {
+      keyEncoding: 'binary',
+      valueEncoding: 'json'
+    })
+  }
+
```

Comment:
In the interest of DRY, maybe have these ``get*()`` helpers call ``this.get()`` instead of ``this._db.get()`` directly? 

``this.get()`` will need to take an optional ``opts`` parameter to handle non-default encodings like in the ``getHeads()`` case